Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,217 @@
+
+using Moq;
+using NUnit.Framework;
+
+using Google.Apis.Auth.OAuth2.Requests;
+using Google.Apis.Auth.OAuth2.Responses;
+using Google.Apis.Http;
+using Google.Apis.Json;
+using Google.Apis.Testing;
+using Google.Apis.Tests;
+using Google.Apis.Util;
+using Google.Apis.Util.Store;
+
+namespace Google.Apis.Auth.OAuth2.DefaultCredentials
+{
+    /// <summary>Tests for <see cref="Google.Apis.Auth.OAuth2.DefaultCredentialProvider"/>.</summary>
+    [TestFixture]
+    public class UserCredentialTests
+    {
+        private string CredentialEnvironmentVariable = "GOOGLE_APPLICATION_CREDENTIALS";
+        private string WellKnownCredentialFilePath = Path.Combine(System.Environment.GetEnvironmentVariable("APPDATA"), "gcloud", "application_default_credentials.json");
+        private const string DummyUserCredentialFileContents = "{\"client_id\": \"CLIENT_ID\",\"client_secret\": \"CLIENT_SECRET\",\"refresh_token\": \"REFRESH_TOKEN\",\"type\": \"authorized_user\"}";
+        private const string DummyServiceAccountCredentialFileContents = "{\"private_key_id\": \"PRIVATE_KEY_ID\",\"private_key\": \"-----BEGIN PRIVATE KEY-----\nMIICdgIBADANBgkqhkiG9w0BAQEFAASCAmAwggJcAgEAAoGBAJJM6HT4s6btOsfe\n2x4zrzrwSUtmtR37XTTi0sPARTDF8uzmXy8UnE5RcVJzEH5T2Ssz/ylX4Sl/CI4L\nno1l8j9GiHJb49LSRjWe4Yx936q0Xj9H0R1HTxvjUPqwAsTwy2fKBTog+q1frqc9\no8s2r6LYivUGDVbhuUzCaMJsf+x3AgMBAAECgYEAi0FTXsu/zRswAUGaViQiHjrL\nuU65BSHXNVjV/2fLNEKnGWGqpli68z1IXY+S2nwbUak7rnGsq9/0F6jtsW+hZbLk\nKXUOuuExpeC5Kd6ngWX/f2jqmhlUabiQijU9cVk7pMq8EHkRtvlosnMTUAEzempu\nQUPwn1PZHhmJkBvZ4lECQQDCErrxl+e3BwUDcS0yVEEmCNSG6xdXs2878b8rzbe7\n3Mmi6SuuOLi3PU92J+j+f/MOdtYrk13mEDdYmd5dhrt5AkEAwPvDEsDT/W4y4h5n\ngv1awGBA5aLFE1JNWM/Gwn4D1cGpEDHKFREaBtxMDCASpHJuw8r7zUywpKhmBZcf\nGS37bwJANdSAKfbafLfjuhqwUJ9yGpykZm/a36aTmerp/bpn1iHdg+RtCzwMcDb/\nTWSwibbvsflgWmHbz657y4WSWhq+8QJAWrpCNN/ZCk2zuGDo80lfUBAwkoVat8G6\nwWU1oZyS+vzIGef+hLb8kHsjeZPej9eIwZ39kcBbT54oELrCkRjwGwJAQ8V2A7lT\nZUp8AsbVqF6rbLiiUfJMo2btGclQu4DEVyS+ymFA65tXDLUuR9EDqJYdqHNZJ5B8\n4Z5p2prkjWTLcA\u003d\u003d\n-----END PRIVATE KEY-----\n\",\"client_email\": \"CLIENT_EMAIL\",\"client_id\": \"CLIENT_ID\",\"type\": \"service_account\"}";
+
+        #region UserCredential
+
+        [Test]
+        public void TestGetApplicationDefaultCredentials_UserCredentials_FromEnvironmentVariable()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, DummyUserCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (!(credential is UserCredential))
+            {
+                Assert.Fail(string.Format("Expected credential of type UserCredential. Obtained {0} instead.", credential.GetType()));
+            }
```

Comment:
We would have to open up multiple layers to encapsulation to check the other properties. How about we address this in a separate checkin.